rust/src/reactive/flat_fat.rs

```diff
@@ -92,7 +90,7 @@ where
     fn new(values: &[Value]) -> Self {
         let capacity = values.len();
         let mut new = Self::with_capacity(capacity);
-        new.update_ordered(values);
+        new.update_ordered(values.iter().cloned());
```

Comment:
Now the values are cloned before passing them in. In reality the iterator is lazy, so it will not clone anything until iteration starts.